src/Cake.Common/Tools/OctopusDeploy/CreateReleaseArgumentBuilder.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Cake.Core;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.OctopusDeploy
+{
+    internal class CreateReleaseArgumentBuilder
+    {
+        private readonly string _projectName;
+        private readonly string _server;
+        private readonly string _apiKey;
+        private readonly CreateReleaseSettings _settings;
+        private readonly ICakeEnvironment _environment;
+
+        private readonly ProcessArgumentBuilder _builder;
+
+        public CreateReleaseArgumentBuilder(string projectName, string server, string apiKey, CreateReleaseSettings settings, ICakeEnvironment environment)
+        {
+            _projectName = projectName;
+            _server = server;
+            _apiKey = apiKey;
+            _settings = settings;
+            _environment = environment;
+            _builder = new ProcessArgumentBuilder();
+        }
+
+        public ProcessArgumentBuilder Get()
+        {
+            AppendCommonArguments();
+
+            AppendArgumentIfNotNull("releaseNumber", _settings.ReleaseNumber);
+            AppendArgumentIfNotNull("defaultpackageversion", _settings.DefaultPackageVersion);
+            AppendPackages(_settings, _builder);
+            AppendArgumentIfNotNull("packagesFolder", _settings.PackagesFolder);
+            AppendArgumentIfNotNull("releasenotes", _settings.ReleaseNotes);
+            AppendArgumentIfNotNull("releasenotesfile", _settings.ReleaseNotesFile);
+
+            if (_settings.IgnoreExisting)
+            {
+                _builder.Append("--ignoreexisting");
+            }
+
+            return _builder;
+        }
+
+        private void AppendCommonArguments()
+        {
+            _builder.Append("create-release");
+
+            _builder.Append("--project");
+            _builder.AppendQuoted(_projectName);
+
+            _builder.Append("--server");
+            _builder.Append(_server);
+
+            _builder.Append("--apiKey");
+            _builder.AppendSecret(_apiKey);
+
+            AppendArgumentIfNotNull("username", _settings.Username);
+
+            if (_settings.Password != null)
+            {
+                _builder.Append("--" + "password");
+                _builder.AppendQuoted(_settings.Password);
```

Comment:
Yes, good catch! Use `AppendQuotedSecret` here to make sure it's not displayed in build logs etc.